lib/types.nix

```
@@ -737,6 +737,118 @@ let
           nestedTypes.elemType = elemType;
         };
 
+      /**
+        TODO: Docs
+      */
+      setRemove = value: {
+        _type = "remove";
+        inherit value;
+      };
+
+      /**
+        TODO: Docs
+      */
+      setWith =
+        { elemType, toKey }:
+        mkOptionType {
+          name = "setOf";
+          description = "set of ${
+            optionDescriptionPhrase (class: class == "noun" || class == "composite") elemType
+          }";
+          descriptionClass = "composite";
+          # disable typeMerging because 'toKey' cannot be merged
+          typeMerge = t: null;
+          check = v: isList v || isType "remove" v;
+          merge =
+            loc: defs:
+            let
+              # Attribute set for efficient key lookup
+              removedKeys = lib.listToAttrs (
+                lib.flatten (
+                  lib.imap1 (
+                    n: def:
+                    if !isType "remove" def.value then
+                      [ ]
+                    else
+                      imap1 (
+                        m: singleDef:
+                        let
+                          optionalValue =
+                            (mergeDefinitions (loc ++ [ "[definition ${toString n}-entry ${toString m}]" ]) elemType [
+                              {
+                                value = singleDef;
+                                file = def.file;
+                              }
+                            ]).mergedValue;
+                        in
+                        if optionalValue ? value then
+                          {
+                            value = optionalValue.value;
+                            name = toKey optionalValue;
```

```suggestion
                            name = toKey optionalValue.value;
```

Make sure the tests catch this :)